r14628 | karl | 2009-08-12 15:31:23 +0200 (Wed, 12 Aug 2009) | 1 line

check for pool overflow in xsubstring; Vincent Danen mail to tex-live, 4 Aug
2009 14:42:29, test case in
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=520920


https://bugzilla.redhat.com/show_bug.cgi?id=492136
https://bugs.gentoo.org/show_bug.cgi?id=264598
CVE-2009-1284

Index: texlive-20080816-source/texk/web2c/bibtex.ch
===================================================================
--- texlive-20080816-source.orig/texk/web2c/bibtex.ch
+++ texlive-20080816-source/texk/web2c/bibtex.ch
@@ -1134,6 +1134,15 @@ while ((ex_buf_xptr < ex_buf_ptr) and
 @y
 @z
 
+% Forgot to check for pool overflow here.  Triggered by test case linked
+% from http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=520920.
+@x 
+while (sp_ptr < sp_end) do                      {shift the substring}
+@y
+str_room(sp_end - sp_ptr);
+while (sp_ptr < sp_end) do                      {shift the substring}
+@z
+
 % [459] Eliminate unreferenced statement label, because `undefined' is
 % now a constant expression that is not evaluated at the Web level. If
 % this label were ever required, it could be replaced by the constant
